src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // now my dreams consist only of split, map, map, filter, filter and reduce...
+  const destcructurizedStr = sourceString.split(';').map(el => el.split(':')
+    .map(
+      space => space.trim()
+    )
+    .filter(
+      zeroValue => zeroValue !== ''
+    ));
```

Comment:
still badly formatted
I'd suggest trimming and filtering inside `reduce`
up to you